experimental/ecma402_listformat/src/lib.rs

```diff
@@ -0,0 +1,554 @@
+/// form, using `Display`.
+pub trait Locale: fmt::Display {}
+
+/// The [listformat] mod contains all the needed implementation bits for `Intl.ListFormat`.
+///
+/// > Note: This is not yet the entire API.  I'd like to get to a consensus on what has been
+/// defined, then use the patterns adopted here for the rest.
+pub mod listformat {
+    /// Contains the API configuration as prescribed by ECMA 402.
+    ///
+    /// The meaning of the options is the same as in the similarly named
+    /// options in the JS version.
+    ///
+    /// See [Options] for the contents of the options.  See the [Format::try_new]
+    /// for the use of the options.
+    pub mod options {
+        /// Chooses the list formatting approach.
+        #[derive(Eq, PartialEq, Debug, Clone)]
+        pub enum Style {
+            Long,
+            Short,
+            Narrow,
+        }
+        /// Chooses between "this, that and other", and "this, that or other".
+        #[derive(Eq, PartialEq, Debug, Clone)]
+        pub enum Type {
+            /// "This, that and other".
+            Conjunction,
+            /// "This, that or other".
+            Disjunction,
+        }
+    }
+
+    /// The options set by the user at construction time.  See discussion at the top level
+    /// about the name choice.  Provides as a "bag of options" since we don't expect any
+    /// implementations to be attached to this struct.
+    ///
+    /// The default values of all the options are prescribed in by the [TC39 report][tc39lf].
+    ///
+    ///   [tc39lf]: https://tc39.es/proposal-intl-list-format/#sec-Intl.ListFormat
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub struct Options {
+        /// Selects a [options::Style] for the formatted list.  If unset, defaults
+        /// to [options::Style::Long].
+        pub style: options::Style,
+        /// Selects a [options::Type] for the formatted list.  If unset, defaults to
+        /// [options::Type::Conjunction].
+        pub in_type: options::Type,
+    }
+
+    /// Allows the use of `listformat::Factory::try_new(..., Default::default())`.
+    impl Default for Options {
+        /// Gets the default values of [Options] if omitted at setup.  The
+        /// default values are prescribed in by the [TC39 report][tc39lf].
+        ///
+        ///   [tc39lf]: https://tc39.es/proposal-intl-list-format/#sec-Intl.ListFormat
+        fn default() -> Self {
+            Options { style: options::Style::Long, in_type: options::Type::Conjunction }
+        }
+    }
+
+    use std::fmt;
+
+    /// The package workhorse: formats supplied pieces of text into an ergonomically formatted
+    /// list.
+    ///
+    /// While ECMA 402 originally has functions under `Intl`, we probably want to
+    /// obtain a separate factory from each implementor.
+    ///
+    /// Purposely omitted:
+    ///
+    /// - `supported_locales_of`.
+    pub trait Format<'a> {
```

Comment:
I'd name it `ListFormat`.

Reply:
Do you need this lifetime? It seems to me that you only use it for `Ret` and `Locale`, which I'm not sure if you need.

Reply:
Done.